### PR TITLE
Planning: added pre_cruise stage for Park_and_Go scenario

### PR DESCRIPTION
**Pull request by sjiang2018**

### CODE REVIEW

File: modules/planning/scenarios/park_and_go/stage_adjust.cc [hunk 1]
```diff
@@ -49,15 +50,34 @@ Stage::StageStatus ParkAndGoStageAdjust::Process(
...
+  if (!ready_to_cruise) {
+    return StageStatus::RUNNING;
+  }
+  if (std::fabs(vehicle_status->steering_percentage()) <
+      scenario_config_.max_steering_percentage_when_cruise()) {
+    next_stage_ = ScenarioConfig::PARK_AND_GO_CRUISE;
+  } else {
+    next_stage_ = ScenarioConfig::PARK_AND_GO_PRE_CRUISE;
+    // reset init position at current position for pre_cruise stage
+    ResetInitPostion(frame);
```
Comment (jmtao) [inline modules/planning/scenarios/park_and_go/stage_adjust.cc]: move  to above code (stage transfer. ie. next_stage_ = ?) to FinishStage() to be consistent and clean.

File: modules/planning/scenarios/park_and_go/stage_pre_cruise.cc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  }
+  const bool ready_to_cruise =
+      scenario::util::CheckADCReadyToCruise(frame, scenario_config_);
+  auto vehicle_status = common::VehicleStateProvider::Instance();
+  ADEBUG << vehicle_status->steering_percentage();
+
+  if (ready_to_cruise &&
+      std::fabs(vehicle_status->steering_percentage()) <
+          scenario_config_.max_steering_percentage_when_cruise()) {
+    return FinishStage();
```
Comment (jmtao) [inline modules/planning/scenarios/park_and_go/stage_pre_cruise.cc]: move this check to prior to CheckADCReadyToCruise() for better performance. 
i.e. fail sooner

File: modules/planning/scenarios/park_and_go/stage_adjust.cc (hunk 1 above)
Comment (sjiang2018) [reply modules/planning/scenarios/park_and_go/stage_adjust.cc]: Fixed. Thank you.

File: modules/planning/scenarios/park_and_go/stage_pre_cruise.cc (hunk 2 above)
Comment (sjiang2018) [reply modules/planning/scenarios/park_and_go/stage_pre_cruise.cc]: Fixed. Thank you.

**Pull request merged**